presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -69,6 +71,7 @@
     private final STRtree rtree;
     private final Map<Integer, Rectangle> partitions;
     private final long memorySizeInBytes;
+    private final LocalMemoryContext localUserMemoryContext;
```

Comment:
This variable is used only in buildRTree. Hence, I suggest to add a parameter to buildRTree and keep buildRTree static and remove this variable.

```
this.rtree = buildRTree(addresses, channels, geometryChannel, radiusChannel, partitionChannel, localUserMemoryContext);

private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> channels, int geometryChannel, Optional<Integer> radiusChannel, Optional<Integer> partitionChannel, LocalMemoryContext localUserMemoryContext)
```